src/scrapers/facebook.iced

```diff
@@ -0,0 +1,117 @@
+    # It's hard to know for sure what tricks an attacker (probably the Keybase
+    # server in this case) might be able to pull with all the different
+    # Facebook endpoints that exist, so we're as strict with the URL here as we
+    # can be. We enforce that post ID's are numeric-only, though we might need
+    # to relax that if we find any exceptions.
+    rxx = new RegExp("^https://m.facebook.com/#{username}/posts/[0-9]+$", "i")
+    return (api_url? and api_url.match(rxx));
+
+  # ---------------------------------------------------------------------------
+
+  check_status: ({username, api_url, proof_text_check, remote_id}, cb) ->
+    # calls back with a v_code or null if it was ok
+    await @_get_url_body { url: api_url }, defer err, rc, raw
+    if err? or rc != v_codes.OK
+      return cb err, rc
+
+    [rc, fb_username, fb_text] = @_extract_username_and_text raw
+    if rc != v_codes.OK
+      return cb null, rc
+
+    if not usernames_equal(username, fb_username)
+      @log "expected username '#{username}' but found '#{fb_username}'"
+      return cb null, v_codes.BAD_USERNAME
+
+    if proof_text_check.trim() != fb_text.trim()
+      @log "expected proof '#{proof_text_check}' but found '#{fb_text}'"
+      return cb null, v_codes.BAD_USERNAME
+
+    cb null, v_codes.OK
+
+  # ---------------------------------------------------------------------------
+
+  _extract_username_and_text : (html) ->
+    $ = @libs.cheerio.load html
+    # Get the username from the first link in the first header in the story.
+    user_profile_link = $('#m_story_permalink_view h3 a').attr('href')
+    if not user_profile_link?
+      @log "failed to find link to author profile"
+      return [v_codes.CONTENT_MISSING, null, null]
+    link_regex = new RegExp("^https://m.facebook.com/([a-zA-Z.]+)?", "i")
+    match = user_profile_link.match(link_regex)
+    if not match?
+      @log "failed to parse author profile link"
+      return [v_codes.CONTENT_MISSING, null, null]
+    username = match[1]
+
+    # Get the proof text from the contents of the second header in the story.
+    proof_text = $('#m_story_permalink_view h3').eq(1).text()
+    if not proof_text? or proof_text == ""
+      @log "failed to find proof text"
+      return [v_codes.CONTENT_MISSING, null, null]
+
+    return [v_codes.OK, username, proof_text]
+
+  # ---------------------------------------------------------------------------
+
+  # Given a validated signature, check that the proof_text_check matches the sig.
+  # TODO: Mostly copied from the Twitter implementation, with some slightly
+  # different whitespace handling. Factor this out?
+  _validate_text_check : ({signature, proof_text_check }) ->
+    [err, msg] = decode_sig { armored: signature }
+    if not err?
+      {short_id} = make_ids msg.body
+      if proof_text_check.split(/\s+/).indexOf(short_id)  < 0
+        err = new Error "Cannot find #{short_id} in #{proof_text_check}"
```

Comment:
Looking at this now. There are a number of places where we support the short ID but not the medium one. Only short is an index in the `signatures` table, and only short is linkable via keybase.io/USER/sigs/SHORT_ID. We could still use the medium ID in the proof text without any changes, if we want, but then it would be a mix of short and medium in different places in the proof / proof flow. Worth it?